biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLQuery.java

```diff
@@ -0,0 +1,270 @@
+package org.biojava.nbio.core.search.io.blast;
+
+
+import org.biojava.nbio.core.search.io.Hit;
+import org.biojava.nbio.core.search.io.Hsp;
+import org.biojava.nbio.core.search.io.Result;
+import org.biojava.nbio.core.search.io.ResultFactory;
+import java.io.File;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import javax.xml.xpath.XPathException;
+import org.biojava.nbio.core.sequence.template.Sequence;
+import org.biojava.nbio.core.util.XMLHelper;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+/**
+ * Re-designed by Paolo Pavan.
+ * You may want to find my contacts on Github and LinkedIn for code info 
+ * or discuss major changes.
+ * https://github.com/paolopavan
+ * 
+ *
+ * @author Scooter Willis <willishf at gmail dot com>
+ * @author Paolo Pavan
+ */
+public class BlastXMLQuery implements ResultFactory {
+    private static final Logger log = Logger.getLogger(BlastXMLQuery.class.getName());
+    Document blastDoc = null;
+    private File targetFile;
+    private List<Sequence> queryReferences, databaseReferences;
+    private Map<String,Sequence> queryReferencesMap, databaseReferencesMap;
+    
+    @Deprecated
+    public BlastXMLQuery(String blastFile) throws Exception {
+        readFile(blastFile);
+    }
+    
+    public BlastXMLQuery() {
+        
+    }
+    public void setFile(File f){
+        targetFile = f;
+    }
+    /**
+     * 
+     * @param maxEScore
+     * @return
+     * @throws Exception 
+     */
+    @Deprecated
+    public LinkedHashMap<String, ArrayList<String>> getHitsQueryDef(double maxEScore) throws Exception {
+        LinkedHashMap<String, ArrayList<String>> hitsHashMap = new LinkedHashMap<String, ArrayList<String>>();

```

Comment:
Thank you Jose. Apart for further comments by @willishf that anyway does not impact on what is done any more, I think that the pull request is ready to be merged. Don't you?
Greetings!
Paolo


> Il giorno 09 ott 2015, alle ore 18:20, Jose Manuel Duarte <notifications@github.com> ha scritto:
> 
> In biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLQuery.java:
> 
> > +    
> > +    public BlastXMLQuery() {
> > +        
> > +    }
> > +    public void setFile(File f){
> > +        targetFile = f;
> > +    }
> > +    /**
> > +     * 
> > +     * @param maxEScore
> > +     * @return
> > +     * @throws Exception 
> > +     */
> > +    @Deprecated
> > +    public LinkedHashMap<String, ArrayList<String>> getHitsQueryDef(double maxEScore) throws Exception {
> > +        LinkedHashMap<String, ArrayList<String>> hitsHashMap = new LinkedHashMap<String, ArrayList<String>>();
> Thanks @paolopavan , the commit looks great! Regarding org.biojava.nbio.genome.query I'd be happy removing it if it really looks like dead code. But I wouldn't want to do something wrong, perhaps @willishf can shed some light on this?
> 
> —
> Reply to this email directly or view it on GitHub.
>